### PR TITLE
KohaRest: Display the date and destination of transitting items

### PR DESCRIPTION
**Pull request by alexklbuckley**
It would be nice if the VuFind search result and record pages showed more details about Koha items with the status of 'In Transit'.

Specifically, it would be helpful to know which library branch the transitting item is going to, and the date it was sent. 

I would love to have some feedback on this please, and specifically it would be great to know where (and if) unit tests are required for this change?

TODO
- [x] Update ILS driver spec when merging
- [x] Merge #3740 to dev before merging this PR

### CODE REVIEW

**Comment by alexklbuckley** [comment]
Fixing the tests

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks for this, too, @alexklbuckley. I think this has exactly the same issues as #3311, so if we resolve our conversation there, we can apply the same approach here.

Also interested in input on @EreMaijala on the Koha-specific aspects of the work.

**Comment by demiankatz** [comment]
Also, regarding unit tests, they're not required but are always appreciated. :-)

**Comment by EreMaijala** [comment]
I'd not change the status message but add a couple of new fields to convey this information in a way that can be displayed as desired. Something like (in_transit is the same as in getMyHolds):

`in_transit` - If in transit, either boolean true or a string describing the destination location; false otherwise 
`in_transit_date` -  If in transit, the timestamp for the event

The do the rest at template level e.g. like with `duedate`.

**Review by EreMaijala — CHANGES_REQUESTED** [review]
I'd not change the status message but add a couple of new fields to convey this information in a way that can be displayed as desired. Something like (in_transit is the same as in getMyHolds):

`in_transit` - If in transit, either boolean true or a string describing the destination location; false otherwise 
`in_transit_date` -  If in transit, the timestamp for the event

The do the rest at template level e.g. like with [due date](https://github.com/vufind-org/vufind/blob/dev/themes/bootstrap3/templates/RecordTab/holdingsils/extended.phtml#L50). `in_transit_date` could also be displayed in parentheses after status.

**Comment by alexklbuckley** [comment]
Thank you very much @demiankatz and @EreMaijala , I will do that - adding a couple of new fields and do the rest at the template level. More patches to come

**Comment by alexklbuckley** [comment]
Hi @demiankatz and @EreMaijala 

I been working on implementing the changes that you suggested in https://github.com/vufind-org/vufind/pull/3312#pullrequestreview-1823399297 and I wonder if I could please have some advice?

Because the getStatusCodeItemTransfer() function is expected to return a string, how best do you suggest I return the in_transit and in_transit_date variables from that function so they can be used in the template?

I have tried the following which doesn't work for the aforementioned limitation of the expected return type of getStatusCodeItemTransfer():

```
$transferDetails = $item['availability']['unavailabilities']['Item::Transfer'];

        $onHold = array_key_exists(
            'Item::Held',
            $item['availability']['notes'] ?? []
        );

        $transfer = [
            'on_hold'  => $onHold,
            'in_transit' => $this->getLibraryName($transferDetails['to_library'] ?? null),
            'in_transit_date => date('d F Y', strtotime($transferDetails['datesent'])),
        ];
        return $transfer;
```

**Comment by EreMaijala** [comment]
@alexklbuckley I wouldn't make that method return all the fields. All it's meant for is to return the status code, so the rest can be handled in getItemStatusesForBiblio just like e.g. `duedate`.

**Comment by alexklbuckley** [comment]
Hi @EreMaijala and @demiankatz 

Apologies for my delay getting updated patches to you. I've just attached the first patch implementing the requested changes. I am working on a follow-up now to fix the failing tests.

Kind regards,
Alex

**Comment by alexklbuckley** [comment]
This is now ready to re-test again @EreMaijala and @demiankatz 

Apologies for the delay!

File: languages/en.ini
Comment (demiankatz) [inline languages/en.ini]: For internationalization, it would be better to use strings with tokens instead of concatenating things in the templates... so, for example, `transit_to` could be changed from `to` to `to %%location%%` and then when you translate it in the template, you could do something like:

`<?=$this->transEsc('transit_to', ['%%location%%' => $item['in_transit'])?>`

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks, @alexklbuckley. I'll let @EreMaijala take the lead on commenting on the KohaRest changes, but in the meantime, I did have a suggestion regarding your new translation strings.

File: themes/bootstrap3/templates/RecordTab/holdingsils/standard.phtml
Comment (EreMaijala) [inline themes/bootstrap3/templates/RecordTab/holdingsils/standard.phtml]: Since in_transit is an optional field, this needs null coalescing (e.g `<?php if ($holding['in_transit'] ?? false): ?>`).

File: themes/bootstrap3/templates/ajax/status-full.phtml
Comment (EreMaijala) [inline themes/bootstrap3/templates/ajax/status-full.phtml]: Null coalescing needed here as well.

File: languages/en.ini
Comment (EreMaijala) [reply languages/en.ini]: Even the `to %%location%%` is problematic in some languages, being a partial sentence. It also assumes that the status is compatible with it, which isn't necessarily the case. I'd suggest something like
`transit_to = "Destination: %%location%%"`
or hiding status and making it
`transit_to = "In transit to %%location%%"`

**Review by EreMaijala — CHANGES_REQUESTED** [review]

**Comment by alexklbuckley** [comment]
Hi @EreMaijala and @demiankatz I have implemented the changes you requested. Could you please take another look now?

File: languages/en.ini
Comment (demiankatz) [inline languages/en.ini]: Minor nitpick, but would a comma or semi-colon after `%%location%%` make this a little more readable?

File: module/VuFind/src/VuFind/ILS/Driver/KohaRest.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/ILS/Driver/KohaRest.php]: Is there any possibility that `$avail['unavailabilities']['Item::Transfer']` might be unset? If so, I'd recommend adding `?? []` to prevent PHP notices; the subsequent isset checks will still yield appropriate behavior if the null coalescing occurs.

File: themes/bootstrap3/templates/RecordTab/holdingsils/standard.phtml
Comment (demiankatz) [inline themes/bootstrap3/templates/RecordTab/holdingsils/standard.phtml]: Is it worth supporting a situation where a transit destination is available but a date is not, in case we implement this for other systems with more limited data availability? I'm thinking something like:

`<?=$this->transEsc(empty($holding['in_transit_date']) ? 'transit_to' : 'transit_to_date', ['%%location%%' => $holding['in_transit'], '%%date%%' => $holding['in_transit_date'] ?? ''])?>`

Then we could define transit_to as a language string without the date, and transit_to_date as the same string with the date added. The code will pick the appropriate string based on the presence or absence of a date value, and the added null coalescing will ensure that a missing date won't cause any errors (it will just be ignored when the 'transit_to' value is used).

File: themes/bootstrap3/templates/ajax/status-full.phtml
Comment (demiankatz) [inline themes/bootstrap3/templates/ajax/status-full.phtml]: If you make my suggested change above, this will of course need to be adjusted to match.

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks, @alexklbuckley! A couple more suggestions designed to make this slightly more robust and flexible.

File: module/VuFind/src/VuFind/ILS/Driver/KohaRest.php
Comment (EreMaijala) [reply module/VuFind/src/VuFind/ILS/Driver/KohaRest.php]: I think it's currently not possible, but it could be if "In Transit" status is set based on something else. I'd actually just remove the `if ($status == 'In Transit')` check, and check instead `if ($transit = $avail['unavailabilities']['Item::Transfer'] ?? null)`

**Review by EreMaijala — CHANGES_REQUESTED** [review]
Just marking this as "Request changes".

**Comment by alexklbuckley** [comment]
Thanks so much for your feedback @EreMaijala , I'll get back to you with changes ASAP

**Comment by alexklbuckley** [comment]
I've pushed some changes, but I need to do additional fixes for the failing tests before this is ready for review.

**Comment by demiankatz** [comment]
Thanks, @alexklbuckley -- just let us know if you need any help figuring out the test problems.

**Comment by alexklbuckley** [comment]
Hi @demiankatz sorry for my delay on this one. Yes if you wouldn't mind giving me a hand with figuring out the test problems that would be much appreciated and hopefully help me unblock this PR!

**Comment by demiankatz** [comment]
@alexklbuckley, the test failure was caused by a parse error in KohaRest.php which in turn appeared to be caused by a missing curly brace. Please look at 9a5fc43; this seems to fix the problem, but I didn't study my work carefully, so it's possible it's not exactly right.

**Comment by alexklbuckley** [comment]
Thank you very much @demiankatz , yes I've had a look at https://github.com/vufind-org/vufind/commit/9a5fc437ad016010f242822529c1343cba419d06 and I believe that is the correct place to put the missing curly brace.

Thanks again!

**Comment by demiankatz** [comment]
Thanks, @alexklbuckley! It looks like there are now some conflicts related to other recent changes. Please let me know if you need help resolving those and/or if this is ready for another review.

**Comment by alexklbuckley** [comment]
Hi @demiankatz ! Thanks very much! I've gone ahead and fixed the conflicts, I noticed themes/bootstrap3/templates/ajax/status-full.phtml has changed quite a lot so I am not 100% sure if I have fixed that conflict correctly or not: https://github.com/vufind-org/vufind/pull/3312/files#diff-ed02f429b1e325fdb886681a362b10a32e93672dc485592360ec76f2ae23e40aR39

I think this is ready for another review.

Thanks so much!

File: module/VuFind/src/VuFind/ILS/Driver/KohaRest.php
Comment (demiankatz) [reply module/VuFind/src/VuFind/ILS/Driver/KohaRest.php]: The `$transit =` assignment following the if statement is not necessary, since the assignment is already accomplished inside the if -- this extra line can be removed without causing any functional changes.

File: themes/bootstrap3/templates/RecordTab/holdingsils/standard.phtml
Comment (demiankatz) [inline themes/bootstrap3/templates/RecordTab/holdingsils/standard.phtml]: I think this code needs to be rewritten a bit -- see lines 20-22 above. This is where the status message and associated class/metadata is set up now, following the refactoring to the availability object. I think instead of having this if/else here, it would make more sense to manipulate those messages earlier -- either by putting conditions in lines 20-22, or by more deeply incorporating this in-transit functionality into the availability object.

**Review by demiankatz — APPROVED** [review]
Thanks, @alexklbuckley. The merging in status-full.phtml looks right to me, but I think standard.phtml needs some more work.

I also left a few other comments.

Note that because this PR adds new language strings, we'll have to wait to merge it until release 10.1 at the earliest, since the window for adding new strings to 10.0 has now passed. Sorry about that!

**Review by demiankatz — CHANGES_REQUESTED** [review]
Sorry, hit the wrong status by accident -- fixing now. :-)

**Comment by alexklbuckley** [comment]
Thanks @demiankatz ! I am working my way through your points now :)

**Comment by demiankatz** [comment]
Thanks, @alexklbuckley -- I've resolved all the conversation threads above that seem to be fully addressed, so it's easier to keep track of where we are. Looks like the last thing remaining is the refactoring of standard.phtml; let me know if I can be of any further help with that (and sorry that this changed so dramatically while we were in the middle of work here...).

**Comment by alexklbuckley** [comment]
Thanks very much for doing that @demiankatz and no problem at all! 

I'm happy to attempt putting conditionals around lines 20-22 for the in-transit functionality to refactor standard.phtml. Would it be at all possible for you to please give me a starting example of how you envisage that could look?

**Comment by demiankatz** [comment]
> Thanks very much for doing that @demiankatz and no problem at all!
> 
> I'm happy to attempt putting conditionals around lines 20-22 for the in-transit functionality to refactor standard.phtml. Would it be at all possible for you to please give me a starting example of how you envisage that could look?

I was thinking something like:

```
      <?php
        if ($holding['in_transit'] ?? null) {
          $statusClass = 'text-danger';
          $statusDescription = empty($holding['in_transit_date']) ? 'transit_to' : 'transit_to_date';
          $statusDescriptionTokens = ['%%location%%' => $holding['in_transit'], '%%date%%' => $holding['in_transit_date'] ?? ''];
          $schemaAvailabilityUri = $availabilityStatus->getSchemaAvailabilityUri();
        } else {
          $statusClass = $this->availabilityStatus()->getClass($availabilityStatus);
          $statusDescription = $availabilityStatus->getStatusDescription();
          $statusDescriptionTokens = [];
          $schemaAvailabilityUri = 'http://schema.org/OutOfStock';
        }
      ?>
```

...and then down below, change `<?=$this->transEsc($statusDescription)?>` to `<?=$this->transEsc($statusDescription, $statusDescriptionTokens)?>`.

**Comment by alexklbuckley** [comment]
Thanks Demian, thank you for that. I have implemented your suggestion, changed the statusDescription line, and removed the original in_transit code blocks. 

I'll fix up this one failing unit test.

**Comment by demiankatz** [comment]
@alexklbuckley, you probably just need to run `vendor/bin/phing update-bootstrap5` and commit the resulting files to clear up the build failure. We introduced the bootstrap5 theme yesterday, so now we need to keep it in sync with bootstrap3 until the older theme can be retired. Fortunately, this phing task should automate most of the work.

**Comment by alexklbuckley** [comment]
Ah righto thanks @demiankatz will do.

**Comment by alexklbuckley** [comment]
That's done the trick, thanks @demiankatz

File: themes/bootstrap3/templates/RecordTab/holdingsils/standard.phtml
Comment (demiankatz) [inline themes/bootstrap3/templates/RecordTab/holdingsils/standard.phtml]: This line should be in the else clause, and the URI from the else clause should be up here. When I was putting together my example, I must have pasted into the wrong clause -- sorry about that!

**Review by demiankatz — CHANGES_REQUESTED** [review]
I see one problem, which is entirely my fault! ;-)

**Comment by alexklbuckley** [comment]
I've fixed up some indentation problems, ready for review again. Thank you!

**Review by demiankatz — CHANGES_REQUESTED** [review]
This looks good to me so far, but it occurs to me that there's likely a need to make some parallel changes to the alternate holdings view in the RecordTab/holdingsils/extended.phtml template. The status handling is very similar to standard.phtml, so it's probably just a copy-and-paste operation. You can test it out via the holdingsTemplate setting in config.ini.

I'm also interested in hearing whether @ThoWagen (who did the recent availability status refactoring) agrees with the approach here or would prefer to see this more tightly integrated into the availability status classes.

**Comment by ThoWagen** [comment]
If I understand this correctly this basically is the same as the STATUS_UNAVAILABLE but with a different status description.

We also have something similar. For example we also have the statuses "lost" or "not_yet_available" for items that have been ordered by the library but that are not available for loan yet. They are treated like sub types of STATUS_UNAVAILABLE. Originally, I just wanted to do implement this as a local customization. What do you think about adding this upstream and making "in_transit" as a sub type of STATUS_UNAVAILABLE? The "in_transit" and "in_transit_to" information could then be added to the status as some kind of "extraStatusInformation". This way, this mechanism could also be user for similar cases.

If you like this idea I would be happy to help with the implementation.

**Comment by demiankatz** [comment]
That sounds reasonable to me, @ThoWagen, and would reduce the number of template changes required here. Perhaps you and @alexklbuckley can work out the best way to move forward (either by him giving you write permission to this branch, or by opening a separate PR to add the facility, and then utilizing it here).

Since this PR adds new language strings and can't be merged until we begin work on 10.1, there is not an enormous rush to complete the work (unless we can somehow introduce some generic mechanisms in 10.0 and then utilize them later here).

**Comment by alexklbuckley** [comment]
Hi @demiankatz and @ThoWagen 

Thank you both for sharing your thoughts on these additional improvements.

You're absolutely understanding correctly Thomas. This enhancement is basically the same as STATUS_UNAVAILABLE but with a different status description.

That's very interesting to hear Thomas about the "lost" and "not_yet_available"  statuses you would like to display. I imagine there are quite a few other libraries in a similar position.

I agree that making "in_transit" a sub set of STATUS_UNAVAILABLE through the addition of some kind of "extraStatusInformation" is a great idea.

It would be great to collaborate with you on the implementation of this. I have given you collaborator rights on this branch  https://github.com/alexklbuckley/vufind/tree/transit_status

What do you see as the first steps? Perhaps you push some changes you have in mind for adding this "extraStatusInformation" addition to the STATUS_UNAVAILABLE then we iterate over them?

**Comment by ThoWagen** [comment]
Maybe the best approach is if I create a separate PR that adds the option to add custom status descriptions. If this change is accepted it should be rather easy to modify this PR and using this feature. And if we discard that idea for some reason can still go back to this PR and merge it as it is.

**Comment by alexklbuckley** [comment]
Yes that sounds like a good idea, thanks @ThoWagen !

**Comment by ThoWagen** [comment]
I created the PR #3740 for the extra status information. It already adds the translations of "transit_to" and "transit_to_date" for testing purposes.

So what would be left to do in this PR is to add the custom status description in the KohaRest driver.

For some reason I missed the fact that having custom status descriptions is already part of VuFind. We only have a custom constructor that simplifies some complexities in creating the status for our own purposes. But it is not suitable for the upstream VuFind.

**Comment by alexklbuckley** [comment]
Hi @ThoWagen thanks very much for that!

I see @demiankatz has provided review requests. What would you like me to contribute - perhaps adding custom status descriptions to the KohaRest driver?

Or maybe something else?

**Comment by ThoWagen** [comment]
@alexklbuckley I think we have to wait for #3740 before moving on with this PR.

**Comment by demiankatz** [comment]
See my recent comment on #3740 for a possible path forward...

**Comment by alexklbuckley** [comment]
Hi @demiankatz and @ThoWagen 

I've merged #3740 into my branch.

Can I please check:

1) Is all I need to do in this branch add the config to the KohaRest.ini, I don't need to make any changes to the KohaRest.php as well?

2) I am a bit confused as to what the configs in KohaRest.ini should look like.  Do they need to go under [[ItemStatusMappings]](https://github.com/vufind-org/vufind/blob/00b5a6281c9cac1bb3baa7ece8367d68bab2f6c3/config/vufind/KohaRest.ini#L150) or elsewhere?

Thanks in advance for your help.

**Comment by ThoWagen** [comment]
@alexklbuckley You don't have to change the KohaRest.ini in order use the new feature from #3740 for your implementation here. It should be sufficient to change the `getItemStatusesForBiblio` method in KohaRest.php to use the new class that represents availability statuses https://github.com/vufind-org/vufind/blob/dev/module/VuFind/src/VuFind/ILS/Logic/AvailabilityStatus.php

Instead of adding both 'availability' and 'status' like this https://github.com/vufind-org/vufind/blob/00b5a6281c9cac1bb3baa7ece8367d68bab2f6c3/module/VuFind/src/VuFind/ILS/Driver/KohaRest.php#L2020-L2021 

you could create an instance of  `AvailabilityStatus`:
```
'avalability' => new AvailabilityStatus($available, $status, ['location' => $inTransit, 'date' => $inTransitDate]);
```

Before this you would have to change `$status` to `transit_to` or `transit_to_date` depending on the case.

I hope this is helpful rather than confusing. Please feel free to ask me to make those changes if you are not sure what to do.

**Comment by alexklbuckley** [comment]
Thanks for that @ThoWagen ! I've pushed a commit to a new branch, not this pull request branch yet.

Is this what you had in mind? https://github.com/alexklbuckley/vufind/commit/c873f125a1eefe53052bc4c660f74df17a2c5e6b

**Comment by ThoWagen** [comment]
@alexklbuckley Yes, you got the idea. 

Unfortunately, there is something wrong with in line 2013. The if statement and a bracket is missing and this should be included after 
```
} else {
    $duedate = null;
}
```

Besides this the 'datesent' part should only be included if 'to_library' exist. Otherwise 'location' in the translation of 'transit_to_date' will be empty.

And finally the 'phpstan-console' is failing because `$inTransit` and `$inTransitDate` might not be defined.

You could fix this by doing something like this:

https://github.com/ThoWagen/vufind/commit/f3e0eb0034e0f05da88fcdfa1e354f82cb59e7a1

**Comment by alexklbuckley** [comment]
Thanks for that @ThoWagen !!

I've made those further changes and pushed my updated branch (with your new feature in https://github.com/vufind-org/vufind/pull/3740 ) merged  to this pull request for review please.

**Comment by ThoWagen** [comment]
I don't have a Koha test environment so I can not verify if this is actually working as expected. But when it comes to using the `AvailabilityStatus` the code looks good to me.

**Comment by demiankatz** [comment]
Thanks, @alexklbuckley! @EreMaijala and I are currently quite busy finishing up last-minute details for the 10.0 release, so reviews may be a bit delayed. If you don't hear from us after the release is done, feel free to send a reminder. :-)

**Comment by alexklbuckley** [comment]
Thanks very much @ThoWagen , @demiankatz and @EreMaijala . 

No problem at all and will do :)

Really appreciate all your assistance!

File: languages/en.ini
Comment (EreMaijala) [inline languages/en.ini]: I think these should be in the HoldingStatus text domain (e.g. languages/HoldingStatus/en.ini).

File: module/VuFind/src/VuFind/ILS/Driver/KohaRest.php
Comment (EreMaijala) [inline module/VuFind/src/VuFind/ILS/Driver/KohaRest.php]: Make this use the same style as the one above (could even combine the two if's but I'm fine with them being separate): `if (null !== ($toLibrary = $transit['to_library'] ?? null)) {`

File: module/VuFind/src/VuFind/ILS/Driver/KohaRest.php
Comment (EreMaijala) [inline module/VuFind/src/VuFind/ILS/Driver/KohaRest.php:2020]: I'm afraid this needs a bit more refactoring so that it doesn't override the choices made above (e.g. pickStatus and the related status rankings).

File: module/VuFind/src/VuFind/ILS/Driver/KohaRest.php
Comment (EreMaijala) [inline module/VuFind/src/VuFind/ILS/Driver/KohaRest.php]: I'm a bit torn on this. On one hand I don't really like it that it's the ILS driver's responsibility to choose the string to display, but on the other hand it's the most flexible approach. But if we know what we want to display, we could just set the final string as the status.

**Review by EreMaijala — CHANGES_REQUESTED** [review]

File: module/VuFind/src/VuFind/ILS/Driver/KohaRest.php
Comment (demiankatz) [reply module/VuFind/src/VuFind/ILS/Driver/KohaRest.php]: Doing things this way (a translation string plus token array) instead of building the whole string here at least makes it a bit more abstract which would simplify some upstream customization scenarios...

File: module/VuFind/tests/integration-tests/src/VuFindTest/Mink/HoldingsTest.php
Comment (ThoWagen) [inline module/VuFind/tests/integration-tests/src/VuFindTest/Mink/HoldingsTest.php]: This needs to be changed to 'HoldingStatus::transit_to'. I already did this in #3740. So you can simply merge the branch into this one again.

File: module/VuFind/src/VuFind/ILS/Driver/KohaRest.php
Comment (ThoWagen) [inline module/VuFind/src/VuFind/ILS/Driver/KohaRest.php]: Since, you moved the translation to  'HoldingStatus.ini' you have to prefix the status with  'HoldingStatus::' here.

File: module/VuFind/src/VuFind/ILS/Driver/KohaRest.php
Comment (ThoWagen) [inline module/VuFind/src/VuFind/ILS/Driver/KohaRest.php]: And you have to add the prefix here as well.

**Review by ThoWagen — DISMISSED** [review]

**Comment by demiankatz** [comment]
I've merged #3740 into dev, and merged dev into this branch, so the changes here are now focused entirely on the KohaRest driver. I'm not 100% sure whether all of @EreMaijala's review requests have been addressed, but it looks likely to me that they have. Since @EreMaijala and @ThoWagen both had "request changes" reviews outstanding, I'm requesting re-reviews from both of them so they can either approve or note any outstanding issues.

Thanks, everyone!

**Comment by alexklbuckley** [comment]
Thank you very much @demiankatz

**Review by ThoWagen — COMMENTED** [review]
My points have been addressed.

File: module/VuFind/src/VuFind/ILS/Driver/KohaRest.php
Comment (EreMaijala) [reply module/VuFind/src/VuFind/ILS/Driver/KohaRest.php:2020]: I believe this is still an issue. E.g. you might want to display "At Bindery" with higher priority than the transfer, but now this overrides the choice made by the pickStatus method. I think that the correct way would be to make pickStatus handle the transfer status too (while taking `$this->statusRankings` into account) and return the extra status information in addition to the chosen status.

**Review by EreMaijala — CHANGES_REQUESTED** [review]

File: module/VuFind/src/VuFind/ILS/Driver/KohaRest.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/ILS/Driver/KohaRest.php]: This is fine as-is, but if you wanted to reduce the code size, you could change this whole block to:

```
if (isset($data['to_library'])) {
    return isset($data['datesent']) ? 'HoldingStatus::transit_to_date' : 'HoldingStatus::transit_to';
}
```

Comment (demiankatz) [reply module/VuFind/src/VuFind/ILS/Driver/KohaRest.php]: A separate question is whether the "on hold" status needs to be combined with the "transit to" status in some way. I see that there's logic related to this below.  Not sure how all the permutations work or what might take priority over what else. Feel free to ignore the question if you don't think this is important -- just asking because I noticed.

File: module/VuFind/src/VuFind/ILS/Driver/KohaRest.php
Comment (demiankatz) [reply module/VuFind/src/VuFind/ILS/Driver/KohaRest.php:2020]: I think this has now been improved, but one more question for @alexklbuckley: do we only want to include the datesent value if `$status` is set to `HoldingStatus::transit_to_date`? While I don't think it hurts anything if we pick a different status and still include the date information, it might be technically more appropriate to only provide it when it applies.

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks for the progress, @alexklbuckley! I suspect this is probably fine to merge as it is, but since @EreMaijala is out of office until August, and I don't want to merge it without his final approval, I've provided a few minor questions and nitpicks in case you think any other improvements are worth making in the meantime.

**Comment by alexklbuckley** [comment]
Hi @demiankatz I don't think there is anything else waiting on me on this pull request is there. We just need to wait for Ere to return now?

**Comment by demiankatz** [comment]
> Hi @demiankatz I don't think there is anything else waiting on me on this pull request is there. We just need to wait for Ere to return now?

I believe that is the case, as long as my questions from the last review don't require any further action.

File: module/VuFind/src/VuFind/ILS/Driver/KohaRest.php
Comment (EreMaijala) [reply module/VuFind/src/VuFind/ILS/Driver/KohaRest.php:2020]: As I mentioned before, I'd like it better if pickStatus method would return also the extra status information, but that could be done in the future. As far as I can see this is now working without side-effects.

**Review by EreMaijala — APPROVED** [review]

File: module/VuFind/src/VuFind/ILS/Driver/KohaRest.php
Comment (demiankatz) [reply module/VuFind/src/VuFind/ILS/Driver/KohaRest.php:2020]: Thanks, @EreMaijala. Since determining the status and the extra status information requires significantly different inputs, I wonder if it would actually make more sense to create a new support method for extra status information instead of overloading pickStatus. In any case, as you say, that refactoring can be done in future as needed (e.g. if the "more status" logic gets longer and more complicated). For now, I'll merge this as-is.

**Review by demiankatz — APPROVED** [review]

**Comment by alexklbuckley** [comment]
thank you very much @demiankatz and @EreMaijala !

**Pull request merged**
